### ISSUE TITLE
Missing git glyphs on WSL

### ISSUE THREAD

**Issue by leesoh** [source-foundry/Hack#369]
Hi there,

First off thanks for the beautiful font! Second, it looks like some glyphs are missing in WSL. Here's a screenshot of a repo:

![missing_glyph](https://user-images.githubusercontent.com/1276654/34906445-377d94b8-f82b-11e7-91ca-7112aea4b73a.png)

I've tried both running the installer, rebooting, and rerunning the installer as well as uninstalling it completely, rebooting, and then reinstalling it. Still no love.

**Comment by chrissimpkins** [comment]
Thanks for the report Liam.  Can I ask you to confirm that this is with the current version of the Windows installer and let us know what glyphs you are expecting to see in those blank areas (ideally with Unicode code points if they are not standard alphanumeric glyphs)?

**Comment by chrissimpkins** [comment]
You can use this tool to identify the Unicode code points https://github.com/source-foundry/uni or refer to the Unicode charts on the web.  If WSL is using glyphs that are not defined by the Unicode standard in the same fashion as the Powerline project, this could be the source of the problem.  I am not familiar with WSL and what you are expecting to see there.

**Comment by leesoh** [comment]
Sorry, guess that would help. I'm just running out - I'll have to dig up the unicode characters when I get back. Here are some screenshots though:

![image](https://user-images.githubusercontent.com/1276654/34907699-cf19da20-f83f-11e7-8fc7-cf37dea8bc90.png)

![image](https://user-images.githubusercontent.com/1276654/34907709-f035b3be-f83f-11e7-8669-b6a2daf03e41.png)


Installer version is 1.5

**Comment by leesoh** [comment]
I should have also mentioned that the X shows up fine when using Deja Vu Sans Mono for Powerline. No luck on the vim things though.

From my screenshots, the ✗ is http://www.fileformat.info/info/unicode/char/2717/index.htm

The ☰is https://codepoints.net/U+2630?lang=en

The ☲ is https://codepoints.net/U+2632?lang=en

**Comment by chrissimpkins** [comment]
@iamjamestl are the stacked horizontal line glyphs still available in your repo? Possible to help him build these into his set? 

We may not have included the 'x' shaped glyph from the DejaVu set.  Will check.   Thanks for posting the additional info!

**Comment by chrissimpkins** [comment]
For my information as I research this issue:

```
CHECK MARK (U+2713) ✓
BALLOT X (U+2717)  ✗
```

**Comment by Sparticuz** [comment]
✖ is missing from mine also build (1709) 16299.192

**Comment by chrissimpkins** [comment]
@Sparticuz Is there a terminal UI style application that uses that glyph on your platform or are you commenting about the glyphs in this code range with check mark and x mark semantics?  Trying to understand the various Powerline like derivatives out there and what type of glyph support is necessary for them.

**Comment by Sparticuz** [comment]
It's from https://stylelint.io/. 

DejaVu sans mono
![image](https://user-images.githubusercontent.com/161913/34995082-79e18cc2-faa3-11e7-90d7-e8eef007a590.png)

Hack
![image](https://user-images.githubusercontent.com/161913/34995110-8f75bb6c-faa3-11e7-9f11-68f507d990d0.png)

**Comment by chrissimpkins** [comment]
@Sparticuz ty! 

We should use this thread as an opportunity to gather these terminal UI style glyphs that are in common use out there.  Anyone aware of others that would be useful to have in the sets?

**Comment by chrissimpkins** [comment]
@leesoh 

I can confirm that we do not have the following glyphs `U+2630`, `U+2632`.  You may need to reach out to @iamjamestl directly to discuss these.  He designed the glyphs in this range for a PR but withdrew the PR with the explantation here https://github.com/source-foundry/Hack/pull/236#issuecomment-345040104 

The following:

```
U+2713 '✓'
U+2716 '✖'
U+2717 '✗'
```

are all glyphs that are not currently included in the Hack sets.  We can definitely consider adding them to support this request. I've updated the OP with a TODO list for these glyphs.

**Comment by chrissimpkins** [comment]
@leesoh Do you happen to have a link to all of the default UI glyphs that are included in Powerline?  They include non-Unicode defined project specific glyphs, but it sounds like there are glyphs from the non-Latin sets of the Unicode standard too.  It would be helpful to know about all of the glyphs that are displayed to users to make sure that we cover this to the degree that we can.

**Comment by leesoh** [comment]
This work? https://github.com/ryanoasis/nerd-fonts

**Comment by chrissimpkins** [comment]
Wtf! Are people actually using trailing flames in their terminal UI? :)

https://github.com/ryanoasis/nerd-fonts#powerline-extra-symbols

![screenshot at jan 22 16-17-13](https://user-images.githubusercontent.com/4249591/35244935-db3346dc-ff8f-11e7-8bf7-da2affe2dae5.png)

This doesn't seem to include any of the Unicode defined code points that we discussed above.

**Comment by chrissimpkins** [comment]
and legos...

**Comment by leesoh** [comment]
If you build it they will come?

**Comment by chrissimpkins** [comment]
Sorry that was insensitive of me.  Clearly, we will not be getting feedback from the flame/lego crowd with those comments. :)  If you are using any the above, let's see some SS.  I want to see how those flames are being used.

**Comment by leesoh** [comment]
This is where I see most of the missing bits (vim-airline): https://2.bp.blogspot.com/-LChtf4Y3trI/V0f3f8uqFaI/AAAAAAAADJM/S1NeL3Ihbs8rHPRt91g0FeBBNQM2pUIoACLcB/w1200-h630-p-k-no-nu/rect4144.png

Here's your flaming lego: https://ryanlmcintyre.com/img/portfolio/powerline-extra-symbols.png

**Comment by chrissimpkins** [comment]
OK I take it back. Does Nerd Fonts patch in airline glyphs?  Might be something to approach Ryan about.  Would be useful to other faces out there through his patching tool.

**Comment by leesoh** [comment]
Yeah, most of the ones that I've noticed missing are in Powerline Extra Symbols.

**Comment by chrissimpkins** [comment]
@leesoh Liam do you see a role for including these symbols here when @ryanoasis already patches them in on the downstream Nerd Fonts package?  Ryan has been keeping up with updates here and offers a patching script to do patch these glyphs in yourself if he misses a release or two along the way and you want to use the changes.  Can you let me know what we are missing c/w what he offers there?

**Comment by chrissimpkins** [comment]
@ryanoasis Ryan if we are missing some of the requested glyphs here, would you be willing to add them to your patching script and support them in the Nerd Fonts builds?  

cc: @leesoh

**Comment by ryanoasis** [comment]
Sure. I am always willing to add glyphs. :smile: Cannot give any ETA though.

We've got some issues with glyphs that I am trying to solve. Nothing Hack specific. Effecting all fonts 

I'll make an issue for this if I remember and if not someone feel free to open.

**Comment by ryanoasis** [comment]
I'll have to check to be completely sure but we don't currently add any sort of glyphs ad hoc. Only ones from sets such as Font Awesome, Octicons

**Comment by leesoh** [comment]
@chrissimpkins Either works for my purposes. If you think it's a better fit elsewhere, no worries. Thanks a lot for following up :)

**Comment by chrissimpkins** [comment]
@leesoh I think the x style glyphs and check are appropriate here.  If Ryan happens to get to them before we do, then we can link to the patched versions for those who need them.  We will try to make this happen.  For the extended Powerline glyphs, I think that Ryan's project is the best source for these patches.  I've worked closely with him on the hints following the patching process and there should not be a significant difference in renders with the downstream builds vs. those upstream.  If there are, please let us know.  

As a side note, @ryanoasis one potential issue for those who use Hack and want your patches is that they get the fonts through package managers.  Are you packaging the Nerd Fonts builds anywhere (Homebrew, Linux distros, etc)?

**Comment by ryanoasis** [comment]
@chrissimpkins Yes actually!

There are [Homebrew Fonts](https://github.com/ryanoasis/nerd-fonts#option-4-homebrew-fonts) and [AUR Packages](https://github.com/ryanoasis/nerd-fonts#option-7-unofficial-arch-user-repository-aur)

**Comment by chrissimpkins** [comment]
@ryanoasis excellent!

**Comment by Dav1dde** [comment]
Any news on this? 

I am on 3.003 (latest from what I can see in the master branch) and it is still missing these two glyphs. Unfortunately nerd-fonts also does not patch them in yet.

**Comment by chrissimpkins** [comment]
@Dav1dde no work towards this yet on our end and I haven't heard any more about this on the Nerd Fonts side.  This work won't happen here until after we push the v4.000 changes.

**Comment by Dav1dde** [comment]
Thanks for the quick response!

If it helps the current "Dejavu Sans Mono Nerd Font Complete" from their git does contain the necessary symbols. So does the latest "Dejavu Sans:style=Book", but not the Mono version!

For anyone with the same problem:
I tried merging these two fonts which kind of worked it messed with the width of the glyphs so I had artifacts between powerline glyphs. Merging partially didn't work for me (probably because I did it wrong). Using the font as fallback works and does look good.

**Comment by rahul0705** [comment]
Been almost a year, v4 being release soon?

**Comment by mynetx** [comment]
Quick status check on my end — this looks less like missing glyphs and more like a terminal font-fallback/cmap issue in WSL/PowerShell/vim specifically. If there's been any progress from maintainers on this, would be good to know.
